front/CODING_RULES.md

```diff
@@ -18,6 +18,14 @@ Favor simple and easy to understand approaches vs overly optimized but complex o
 Reviewer: If you detect an overly optimized or complex solution that can be simplified (at the cost
 of a bit of performance loss or extra code), ask the author to consider the simpler approach.
 
+### [GEN3] Favor types over typescript enums
+
+We do not use typescript enums, we use types instead, eg: `type Color = "red" | "blue";`.
+
+### [GEN4] Use of `as` is prohibited
+
+The use of `as` is prohibited in the codebase. Use typeguards or other type-safe methods instead.
```

Comment:
It can't be fully prohibited though. You need it when you reduce into an object (and it's type safe in those cases), you need it for "as const", and there are many cases where `as` is totally fine and type safe

That said, aligned that in most cases typeguards should be used instead of as

Reply:
Will add these exceptions :+1:

Reply:
What do you mean by `reduce into an object` ?